source/MRMesh/MRPolylineTrimWithPlane.h

```diff
@@ -1,22 +1,27 @@
 #pragma once
-#pragma once
 #include "MRMeshFwd.h"
+#include "MRBitSet.h"
 
 namespace MR
 {
 /// This function splits edges intersected by the plane
-/// \return New edges with origin on the plane and oriented to the positive direction
+/// \return edges located above the plane (in direction of normal to plane)
 /// \param polyline Input polyline that will be cut by the plane
 /// \param plane Input plane to cut polyline with
+/// \param newPositiveEdges new edges with origin on the plane and oriented to the positive direction
 /// \param onEdgeSplitCallback is invoked each time when an edge is split. Receives edge ID before split, edge ID after split, and weight of the origin vertex
-MRMESH_API EdgeBitSet subdividePolylineWithPlane( Polyline3& polyline, const Plane3f& plane, std::function<void( EdgeId, EdgeId, float )> onEdgeSplitCallback = nullptr );
+MRMESH_API EdgeBitSet subdivideWithPlane( Polyline3& polyline, const Plane3f& plane, EdgeBitSet* newPositiveEdges = nullptr, std::function<void( EdgeId, EdgeId, float )> onEdgeSplitCallback = nullptr );
+[[deprecated]] inline EdgeBitSet subdividePolylineWithPlane( Polyline3& polyline, const Plane3f& plane, std::function<void( EdgeId, EdgeId, float )> onEdgeSplitCallback = nullptr )
+{
+    subdivideWithPlane( polyline, plane, nullptr, onEdgeSplitCallback );
+}
 
 struct DividePolylineParameters
 {
     /// onEdgeSplitCallback is invoked each time when an edge is split. Receives edge ID before split, edge ID after split, and weight of the origin vertex
     std::function<void( EdgeId, EdgeId, float )> onEdgeSplitCallback;
-    /// fillAfterCut if true, the ends of resulting polyline will be united with new edges
-    bool fillAfterCut = false;
+    /// closeLineAfterCut if true, the ends of resulting polyline will be united with new edges
```

Comment:
better describe two options:
1) plane cuts closed line or even time open line
2) plane cuts open line once (here no new edge must be created)

Reply:
expanded comment (not really sure about this)